app/assets/javascripts/discourse/app/components/flag-action-type.js

```diff
@@ -17,14 +17,9 @@ export default Component.extend({
     return I18n.t("flagging.custom_placeholder_" + nameKey);
   },
 
-  @discourseComputed(
-    "flag.name",
-    "flag.name_key",
-    "flag.is_custom_flag",
-    "username"
-  )
-  formattedName(name, nameKey, isCustomFlag, username) {
-    if (isCustomFlag) {
+  @discourseComputed("flag.name", "flag.name_key", "username")
+  formattedName(name, nameKey, username) {
+    if (["notify_user", "notify_moderators"].includes(nameKey)) {
       return name.replace(/{{username}}|%{username}/, username);
     } else {
       return I18n.t("flagging.formatted_name." + nameKey);
```

Comment:
The flag popup translation for "illegal" will now go through this code path, which looks in the `client.en.yml` file. This is so we can have "it's illegal" in the popup, but "is this post illegal?" in the review queue.